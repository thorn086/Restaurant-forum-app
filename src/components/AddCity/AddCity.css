.city-form{
    border: 1px solid black;
    width: 90%;   
    background-color: rgb(70, 70, 193);
    margin:1rem;
    border-radius: 5%;
    box-shadow: 6px 6px rgb(61, 61, 95);
    padding: 1rem 1rem 1.5rem 1rem;
    text-align: center;
    height: auto;
    letter-spacing: 2px;
    color: white;
    font-size: 16px;
    font-weight: 600;
    line-height: 2;
    }
.city_field{
    margin-top: 2rem;
}
.back-btn.city_form{
    margin: 0;
}
#name{
    width: 12rem;
    height: 1.5rem;
    margin-bottom: 2rem;
}
@media (min-width: 377px){
    .city-form{
        width: 90%;
        margin: 1rem auto;
    }
    #name{
        width: 15rem;
    }
    .back-btn.city_form{
        margin:1rem auto;
    }
}   
@media (min-width: 431px){
    .city-form{
        width: 80%;
        margin: 1rem auto;
    }
    #name{
        width: 17rem;
    }
    
}  
@media (min-width: 539px){
    .city-form{
        width: 80%;
        margin: 1rem auto;
    }
    #name{
        width: 20rem;
    }
    .back-btn.city_form{
        margin:1rem auto;
    }
}  
@media (min-width: 650px){
    .city-form{
        width: 60%;
        margin: 1rem auto;
        font-size: 18px;
    }
    #name{
        width: 24rem;
    }
    .back-btn.city_form{
        margin:1rem auto;
    }
}  
@media (min-width: 912px){
    .city-form{
        width: 50%;
        margin: 1rem auto;
        font-size: 18px;
    }
    #name{
        width: 26rem;
    }
    .back-btn.city_form{
        margin:1rem auto;
    }
} 
@media (min-width: 1226px){
    .city-form{
        width: 60%;
        margin: 1rem auto;
        font-size: 18px;
    }
    #name{
        margin: 1rem;
        width: 30rem;
    }
    .back-btn.city_form{
        margin:1rem auto;
    }
} 
