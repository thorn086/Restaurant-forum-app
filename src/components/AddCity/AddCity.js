import React from 'react';
import config from '../../config';
import { withRouter } from 'react-router-dom';
import TokenService from '../../services/token-services';
import restForumContext from '../../context';
import './AddCity.css';

class AddCity extends React.Component {
    static defaultProps = {
        history: {
            push: () => {}
        },
        match: {
            params: {}
        }
    }

    static contextType = restForumContext

   


    handleSubmit = event => {
        event.preventDefault()
        const { id } = this.props.match.params;
        const newCity = {
            name: event.target['city-input'].value,
            state_id: id,
            author: `${TokenService.getUserId()}`
        };
        //adds new city to selected state
        fetch(`${config.API_ENDPOINT}/states/${id}`, {
            method: 'POST',
            headers: {
                'content-type': 'application/json',
                'authorization': `bearer ${TokenService.getAuthToken()}`
            },
            body: JSON.stringify(newCity),
        })
            .then(results => {
                if (!results.ok)
                    return results.json().then(e => Promise.reject(e));
                return results.json();
            })
            .then(city => {
                this.context.addCity(city);
                this.props.show();
                this.props.history.push(`/states/${id}`);
                
            })
            .catch(error => {
                console.error({ error });
            })
    }


    render() {
        
        return (
            <form className="city-form" onSubmit={this.handleSubmit}>
                <div className='city_field'>
                    <label htmlFor='city_name'>City Name</label>
                    <input type='text' id='name' name='city-input' placeholder='City Name' required></input>
                </div>
                <div className='author'>
                    <p>City added by: {TokenService.getUserId()}</p>
                </div>

                <button type='submit' className='back-btn city_form'>Add</button>
            </form>
        )
    }
}
export default withRouter(AddCity)