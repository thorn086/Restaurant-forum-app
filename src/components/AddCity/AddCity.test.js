import React from 'react';
import AddCity from './AddCity';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('AddCity', () => {
    const addCity = shallow(<AddCity/>);
  
    it('renders properly', () => {
        expect(addCity).toMatchSnapshot();
    });
});