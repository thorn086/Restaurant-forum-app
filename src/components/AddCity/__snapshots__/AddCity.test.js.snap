// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCity renders properly 1`] = `ShallowWrapper {}`;
